fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/TypeRepository.java

```diff
@@ -435,13 +433,15 @@ private void addEnumType(@Nonnull final EnumDescriptor enumType,
     public static class Builder {
         private @Nonnull final FileDescriptorProto.Builder fileDescProtoBuilder;
         private @Nonnull final FileDescriptorSet.Builder fileDescSetBuilder;
-        private @Nonnull final BiMap<Type, String> typeToNameMap;
+        private @Nonnull final Map<Type, String> typeToNameMap;
+        private @Nonnull final Map<String, Type> nameToCanonicalTypeMap;
```

Comment:
Is there a reason this need to go from a `BiMap` to two `Map`s? The `BiMap` should already be encapsulating that kind of thing, and it has an `inverse()` method if you need to explicitly have access to a map-like thing that goes the other way.